index.html

```diff
@@ -10328,47 +10447,72 @@ <h2>User prompts</h2>
 
  <dl>
   <dt>"<code>text</code>"
-  <dd>The <a>current user prompt</a>’s
+  <dd>The <a>current user prompt</a>&apos;s
    <a data-lt="user prompt message">message</a>.
  </dl>
 
-<p>To <dfn>handle any user prompts</dfn>
- given <var>session</var>, <a>remote end</a> must:
+<p>To <dfn>get the prompt handler</dfn> <var>type</var>:
 
-<ol>
-<li><p>
-If there is no <a>current user prompt</a>,
-abort these steps and return <a>success</a>.
+<ol class=algorithm>
+ <li><p>If the <a>user prompt handler</a> is null,
+ let <var>handlers</var> be an empty map. Otherwise
+ let <var>handlers</var> be <a>user prompt handler</a>.
 
-<li><p>
-Perform the following substeps
-based on the <a>session</a>’s <a>user prompt handler</a>:
+ <li><p>If <var>handlers</var> contains <var>type</var>
+ return <var>handlers</var>[<var>type</var>].
 
-<dl class=switch>
-<dt><a>dismiss state</a>
-<dd><p><a>Dismiss</a> the <a>current user prompt</a>.
+ <li><p>If <var>type</var> is "<code>beforeUnload</code>", return a
```

Comment:
I assume that we can move this up given that it doesn't depend on any user prompt handler setting.

Reply:
Ah good point. Would it make sense to add a note here which explains why it needs to stay here?

Reply:
I added this to the note above.